install-deps.sh

```diff
@@ -146,21 +146,34 @@ function install_pkg_on_ubuntu {
     fi
 }
 
+boost_ver=1.82
+boost_installed_ver=$(apt -qq list --installed 'ceph-libboost*-dev' 2> /dev/null \
+			  | grep -e 'libboost[0-9].[0-9]\+-dev' \
+			  | cut -d ' ' -f 2 \
+			  | cut -d '.' -f1,2)
+
+function clean_boost_on_ubuntu {
+    local ver=$boost_ver
+    local installed_ver=$boost_installed_ver
+    $SUDO rm -f "/etc/apt/sources.list.d/ceph-libboost${installed_ver}.list"
+    $SUDO rm -f "/etc/apt/sources.list.d/ceph-libboost.list"
+    $SUDO rm -f "/etc/apt/sources.list.d/libboost.list"
+    $SUDO apt-get update -y
+    apt -qq list --installed 'ceph-libboost*-dev' 2> /dev/null | \
+	grep -F -v "$boost_ver" | \
+	cut -f 1 -d / | \
+	xargs "$SUDO" apt-get -y remove
+}
+
 function install_boost_on_ubuntu {
     ci_debug "Running install_boost_on_ubuntu() in install-deps.sh"
-    local ver=1.82
-    local installed_ver=$(apt -qq list --installed ceph-libboost*-dev 2>/dev/null |
-                              grep -e 'libboost[0-9].[0-9]\+-dev' |
-                              cut -d' ' -f2 |
-                              cut -d'.' -f1,2)
+    local installed_ver=$boost_installed_ver
     if test -n "$installed_ver"; then
         if echo "$installed_ver" | grep -q "^$ver"; then
```

Comment:
This check seems broken as `$ver` variable definition has been moved.

Reply:
Fixed